string unique_id

base base_component 
arm arm_component 

